### DISCUSSION TITLE
OpenAPI - query params are not shown correctly with additional Depends in a route

### DISCUSSION THREAD

**Discussion by AntonKyrychekCrit** [fastapi/fastapi#12265] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated, Literal

from fastapi import FastAPI, Query, Depends
from pydantic import BaseModel, Field

app = FastAPI()


class FilterParams(BaseModel):
    limit: int = Field(100, gt=0, le=100)
    offset: int = Field(0, ge=0)
    order_by: Literal["created_at", "updated_at"] = "created_at"
    tags: list[str] = []


def extra_depends_func(limit: int) -> str:
    return 'does something'


@app.get("/items/")
async def read_items(
    filter_query: Annotated[FilterParams, Query()],
    extra_depends: str = Depends(extra_depends_func),
):
    return filter_query
```

### Description

My project has a bunch of depends for multiple routers and we would like to use a new feature that came in 0.155.0 for autodocs.
I looked at https://fastapi.tiangolo.com/tutorial/query-param-models/ and after trying to implement  it, it gives back this 
![image](https://github.com/user-attachments/assets/40907736-f8bd-4c0e-b693-20c4bc985703)

I understand that extra_depends_func has `limits` param and that's what is causing a problem. 
I'm not sure it this is a bug or intended or is there a way to bring filter_query up from the nested object so it would look like in the tutorial for this feature?

### Operating System

Linux

### FastAPI Version

0.115.0

### Pydantic Version

2.8.2

### Python Version

3.12.0

**Comment by Kfir-G** [comment]
The issue here is that both your `FilterParams` model and the `extra_depends_func` dependency are using the same parameter name, `limit`. This creates a conflict in FastAPI, which results in the query parameters not displaying correctly in the OpenAPI documentation. You can learn more about this in the FastAPI docs [here](https://fastapi.tiangolo.com/tutorial/dependencies/#dependencies).

**Reply by AntonKyrychekCrit** [reply]
So there is no way it can be resolved? We have a lot of shared and complex logic with Depends on the level of routers.
and that link you provided has nothing on how custom depends effect OpenAPI docs

**Reply by Kfir-G** [reply]
Hey @AntonKyrychekCrit ,

When using dependencies with overlapping parameter names, FastAPI can struggle with displaying the query parameters correctly in the OpenAPI docs. Unfortunately, this is a limitation in how FastAPI resolves dependencies.

One way to resolve this is to rename the parameter in either your FilterParams model or the extra_depends_func to avoid the conflict. For example, you could change limit in extra_depends_func to something like extra_limit.

While I get that this might not be ideal if you have complex shared logic, it helps clarify the parameter definitions. If you're relying heavily on shared logic, consider wrapping it in a separate function that handles the common parameters without naming conflicts.

**Comment by YuriiMotov** [comment]
Duplicate of https://github.com/fastapi/fastapi/discussions/12212.
Let's close this topic and track it there
